hypha/apply/projects/views/payment.py

```diff
@@ -162,22 +196,40 @@ def dispatch(self, request, *args, **kwargs):
 
     def get_initial(self):
         initial = super().get_initial()
-
         initial["supporting_documents"] = [
             document.document for document in self.object.supporting_documents.all()
         ]
         return initial
 
     def form_valid(self, form):
         response = super().form_valid(form)
-
-        messenger(
-            MESSAGES.UPDATE_INVOICE,
-            request=self.request,
-            user=self.request.user,
-            source=self.object.project,
-            related=self.object,
-        )
+        if form.changed_data:
+            self.object.status = 'Resubmitted'
+            self.object.save()
+
+            messenger(
+                MESSAGES.UPDATE_INVOICE,
+                request=self.request,
+                user=self.request.user,
+                source=self.object.project,
+                related=self.object,
+            )
+
+            if 'message_for_pm' in form.changed_data:
+                invoice_status_change = _(f'<p>Invoice status updated to : {self.object.status_display}. </p>')
```

Comment:
F-strings and translations issue here. Use format().

Reply:
This seems to have been reverted by mistake.

Reply:
While you are at it you can remove the extra space before the colon as well, here and in one or two other places as well.

Reply:
As well as space after the last period. It have no effect on the output but un neat stuff like that bothers me :-).